pkg/machinery/config/types/v1alpha1/v1alpha1_types.go

```diff
@@ -1051,6 +1069,45 @@ type DiskPartition struct {
 	DiskMountPoint string `yaml:"mountpoint,omitempty"`
 }
 
+// EncryptionConfig represents partition encryption settings.
+type EncryptionConfig struct {
+	//   description: >
+	//     Encryption provider to use for the encryption.
+	//   examples:
+	//     - value: '"luks2"'
+	EncryptionProvider string `yaml:"provider"`
+	//   description: >
+	//     Defines the encryption keys generation and storage method.
+	EncryptionKeys []*EncryptionKey `yaml:"keys"`
+	//   description: >
+	//     Cipher kind to use for the encryption.
+	//     Depends on the encryption provider.
+	EncryptionCipher string `yaml:"cipher,omitempty"`
+}
+
+// EncryptionKey represents configuration for disk encryption key.
+type EncryptionKey struct {
+	//   description: >
+	//     Key which value is stored in the configuration file.
+	KeyStatic *EncryptionKeyStatic `yaml:"static,omitempty"`
+	//   description: >
+	//     Deterministically generated key from the node UUID and PartitionLabel.
+	KeyGenerated *EncryptionKeyGenerated `yaml:"generated,omitempty"`
+	//   description: >
+	//     Key slot number for luks2 encryption.
+	KeySlot int `yaml:"slot"`
+}
+
+// EncryptionKeyStatic represents throw away key type.
+type EncryptionKeyStatic struct {
+	//   description: >
+	//     Defines the static key value.
+	KeyData string `yaml:"key,omitempty"`
+}
+
+// EncryptionKeyGenerated represents deterministically generated key from the node UUID and PartitionLabel.
+type EncryptionKeyGenerated struct{}
```

Comment:
Same as above, probably need some better name

Reply:
I was asking for a better name on the review, but haven't got any good suggestions :)